### DISCUSSION TITLE
第2节课笔记(7班）

### DISCUSSION THREAD

**Discussion by JimmyMa99** [InternLM/Tutorial#84] [General]
第2节课笔记(7班）

**Comment by abelezm** [comment]
第二课：《轻松玩转书生·浦语大模型趣味 Demo》
主讲：
课堂实验过程：
       基本顺利，按照教程step by step，非常顺利，很震撼的感觉，木有想到速度这么快。不知道是模型还是A100的问题。数学题，物理题都可以很快的得出结果，强。
https://blog.csdn.net/weixin_38043453/article/details/135437893

ssh -CNg -L 127.0.0.1:6006:0.0.0.0:6006 root@ssh.intern-ai.org.cn -p 33693

**Comment by victory3398** [comment]
第二课: 《轻松玩转书生浦语大模型趣味Demo》
按照文档内容，基本实现了InternLM-Chat-7B 智能对话Demo、Lagent 智能体工具调用Demo、浦语灵笔图文理解创作Demo。最后一个图文理解Demo加载过程遇到点困难文理解Demo加载过程遇到点困难。

**Comment by createup2023** [comment]
笔记：https://www.yuque.com/creatup/ynrsde/qatzarnug4muz602?singleDoc# 《第二节课》
补充CSDN笔记：https://blog.csdn.net/weixin_48339493/article/details/135756364?spm=1001.2014.3001.5502

**Comment by wuhaostudio** [comment]
https://blog.csdn.net/2301_77286822/article/details/135433229

**Comment by YangXuanyi** [comment]
## 第二节课笔记
第二节课主要以运行书生·浦语大模型趣味demo的形式讲解了如何调用大模型。具体讲解了三个demo的使用方法和通用环境的配置方法。
### 1 InternLM-Chat-7B 智能对话 Demo
这部分以终端和web两种运行方式讲述了如何从0开始调用大模型，涉及到了开发机的环境配置、模型下载、代码准备等。
### 2 Lagent 智能体工具调用 Demo
这部分讲述了如何快速实现一个智能体demo，相较于上一节的通用大模型来说，智能体支持用户快速制定某种类型的工具，针对性更强。
### 3 浦语·灵笔图文理解创作 Demo
图文理解demo可以用来生成图文并茂的博客文章。

### 4 通用环境配置
这部分讲解了一些关于使用大模型的基础环境配置的方法，包括换源，配置端口和模型下载等内容，是对前面demo运行中涉及到的基础环境配置的集中讲解。

### 总结
第二节课从代码的角度讲解了怎么调用模型，但是对那些demo程序和大模型内部的程序没有深入的讲解。我希望在未来的课程能多安排一些代码讲解，而不是直接粘贴和运行代码。

**Comment by PommesPeter** [comment]
## 第二节课笔记

博客链接：[https://memo.sylin.host](https://memo.sylin.host/docs/DeepLearning/%E5%9F%BA%E7%A1%80%E7%9F%A5%E8%AF%86/%E8%AF%BE%E7%A8%8B%E5%AD%A6%E4%B9%A0/%E4%B9%A6%E7%94%9F%C2%B7%E6%B5%A6%E8%AF%AD%E5%A4%A7%E6%A8%A1%E5%9E%8B%E5%AE%9E%E6%88%98/%E7%AC%AC%E4%BA%8C%E8%8A%82%E8%AF%BE%E7%AC%94%E8%AE%B0)

### 简介

- InternLM 是一个开源的轻量级训练框架，旨在支持大模型训练而无需大量的依赖。基于InternLM训练框架，上海人工智能实验室已经发布了两个开源的预训练模型: InternLM-7B 和 InternLM-20B。
- Lagent 是一个轻量级、开源的基于大语言模型的智能体(agent)框架，用户可以快速地将一个大语言模型转变为多种类型的智能体。通过 Lagent 框架可以更好的发挥 InternLM 模型的全部性能。
- 浦语·灵笔是基于书生·浦语大语言模型研发的视觉·语言大模型，有着出色的图文理解和创作能力，使用浦语·灵笔大模型可以轻松的创作一篇图文推文。

### InternLM-Chat-7B 对话
InternLM-7B 包含了一个拥有 70 亿参数的基础模型和一个为实际场景量身定制的对话模型。该模型具有以下特点:

1. 利用数万亿的高质量 token 进行训练，建立了一个强大的知识库。
2. 支持 8k token 的上下文窗口长度，使得输入序列更长并增强了推理能力。

### Lagent
Lagent 是一个轻量级、开源的基于大语言模型的智能体 (agent) 框架，用户可以快速地将一个大语言模型转变为多种类型的智能体，并提供了一些典型工具为大语言模型赋能。架构如图所示:

![image](https://github.com/InternLM/tutorial/assets/54879512/9bfb3453-afbd-45d7-899b-81ea64e930f4)

### InternLM-XComposer-7B
浦语·灵笔是基于书生·浦语大语言模型研发的视觉-语言大模型，提供出色的图文理解和创作能力，具有多项优势:

1. 为用户打造图文并貌的专属文章。
2. 设计了高效的训练策略，为模型注入海量的多模态概念和知识数据，赋予其强大的图文理解和对话能力。

## 总结 & 思考
通过全链路的开发体系，减少了大模型应用的门槛，同时，框架对训练进行优化使得训练大模型不再遥远。以后希望通过 InternLM 体系开发一个属于个人的 AI 助手。

**Comment by hui1feng** [comment]
第二节笔记：https://github.com/hui1feng/-Demo/blob/main/README.md

**Comment by anneeeee** [comment]
笔记：https://mp.weixin.qq.com/s/UTh8DJepnP5R2P1WnFO83A

**Comment by SaaRaaS-1300** [comment]
第2堂课的笔记及作业链接：github-> https://github.com/SaaRaaS-1300/InternLM_openNotebook/blob/main/Lesson-2/Lesson-2-Notebook.md

**Comment by thefengl** [comment]
## 基础作业

使用 InternLM-Chat-7B 模型生成 300 字的小故事（需截图）

![浦语大模型趣味Demo_1](https://github.com/InternLM/tutorial/assets/83425837/76a87fed-8203-4574-aa82-20479f500a6f)

熟悉 hugging face 下载功能，使用 huggingface_hub python 包，下载 InternLM-20B 的 config.json 文件到本地（需截图下载过程）。

![浦语大模型趣味Demo_2](https://github.com/InternLM/tutorial/assets/83425837/4cb661a8-a122-480b-9a8e-54fc6f09a459)

## 进阶作业

完成浦语·灵笔的图文理解及创作部署（需截图）

![浦语大模型趣味Demo_5](https://github.com/InternLM/tutorial/assets/83425837/3616d76c-85eb-46cd-b979-4616da26a050)

![浦语大模型趣味Demo_6](https://github.com/InternLM/tutorial/assets/83425837/60c2794a-0f6c-49c2-a915-796290b4a6ee)

完成 Lagent 工具调用 Demo 创作部署（需截图）

![浦语大模型趣味Demo_3](https://github.com/InternLM/tutorial/assets/83425837/eb4695b2-b427-498b-b6ae-a6b1e527750b)

![浦语大模型趣味Demo_4](https://github.com/InternLM/tutorial/assets/83425837/b0dfe780-cf76-4911-97c5-38b0a8eb5d8d)

**Comment by lifelinezhang** [comment]
https://zhuanlan.zhihu.com/p/676476967

**Comment by snowhou** [comment]
第二课笔记打卡：https://zhuanlan.zhihu.com/p/677131304?

**Comment by leochiang** [comment]
第二节笔记 https://vvgyb242x83.feishu.cn/wiki/O8vKwSwesiHGvTk2r4gcSGWDndd?from=from_copylink

**Comment by Simon-lyz** [comment]
https://g4w90egc30.feishu.cn/docx/KIw4ddOrQoIeWCxJMjbciXb8nbf?from=from_copylink

**Comment by Rogue-Trader-zzy** [comment]
第二节课笔记：https://zhuanlan.zhihu.com/p/680476195
